### ISSUE TITLE
Queue#seek method working very bad

### ISSUE THREAD

**Issue by Olebeh** [Androz2091/discord-player#1303]
First, it seeks so slow, if you want to seek for 10 seconds forward, you should wait 30 seconds to seek only TEN. Second, using it with await doesn't work quite right. The promise resolves before the track actually seeks. I don't know if the first problem can be fixed, but i hope the second can be

**Comment by DevAndromeda** [comment]
> First, it seeks so slow, if you want to seek for 10 seconds forward, you should wait 30 seconds to seek only TEN. Second, using it with await doesn't work quite right. The promise resolves before the track actually seeks. I don't know if the first problem can be fixed, but i hope the second can be

Seek works just like the audio filters. It will be fast if the pulled stream starts buffering quickly. This is definitely the issue I would like to improve and is on my todo list for v6 update.

**Comment by Nitamet** [comment]
Until the slow seek problem is fixed, it would be great if we could at least determine the moment when a track actually starts playing after seek(), because the promise indeed resolves long before that. Haven't read source code well, but maybe we can emit an event when the stream is created?

**Comment by skdhg** [comment]
> Until the slow seek problem is fixed, it would be great if we could at least determine the moment when a track actually starts playing after seek(), because the promise indeed resolves long before that. Haven't read source code well, but maybe we can emit an event when the stream is created?

Yeah that is possible

**Comment by Nitamet** [comment]
Okay, so I did some research, and actually to get the exact moment when the track starts playing we need not the moment when the stream is created, but when the connection `start` event is emitted. In its listener, we already emit event `trackStart` if `filtersUpdate` is set to false. In the case of `seek()` it is set to true, and no event is fired. In general, there is no event for cases where `filtersUpdate` is set to true. So I added a new event `trackUpdate` in PR #1543 .
